pkg/processor/jsprocessor/jsprocessor.go

```diff
@@ -0,0 +1,184 @@
+// Copyright © 2022 Meroxa, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package jsProcessor
+
+import (
+	"context"
+
+	"github.com/conduitio/conduit/pkg/foundation/cerrors"
+	"github.com/conduitio/conduit/pkg/processor"
+	"github.com/conduitio/conduit/pkg/record"
+	"github.com/dop251/goja"
+	"github.com/rs/zerolog"
+)
+
+const (
+	entrypoint = "process"
+)
+
+// jsProcessor is able to run transformations defined in JavaScript.
+type jsProcessor struct {
+	runtime  *goja.Runtime
+	function goja.Callable
+}
+
+func NewJSProcessor(src string, logger zerolog.Logger) (processor.Processor, error) {
```

Comment:
A couple of nitpicks here:
- Keep in mind that every call to this package will contain the package name, so `jsProcessor.NewJSProcessor` sounds like stuttering (obligatory [reference link](https://github.com/golang/go/wiki/CodeReviewComments#package-names)).
- The general rule of thumb is to accept interfaces and return concrete types, meaning that here it would be better to not hide `jsProcessor` behind `processor.Processor`, but rather to export `jsProcessor` and return the concrete type, letting the caller decide if they want to use it through the `processor.Processor` interface.